src/tests/integration/IntegrationBase.t.sol

```diff
@@ -0,0 +1,939 @@
+        // Lower the debt ceiling to a small number
+        uint256 debtCeiling = uint256(standardDebtSize / 2);
+        vm.prank(admin); vat.file(ilk, "line", debtCeiling * RAY);
+
+        // Max out the debt ceiling
+        setDebtToMaximum();
+        (uint256 ink, uint256 art) = vat.urns(ilk, address(pool));
+        assertEq(ink, debtCeiling);
+        assertEq(art, debtCeiling);
+        assertRoundingEq(getLPTokenBalanceInAssets(address(pool)), debtCeiling);
+
+        // Should be a no-op
+        hub.exec(ilk);
+        (ink, art) = vat.urns(ilk, address(pool));
+        assertEq(ink, debtCeiling);
+        assertEq(art, debtCeiling);
+        assertRoundingEq(getLPTokenBalanceInAssets(address(pool)), debtCeiling);
+
+        // Raise it by a bit
+        debtCeiling = uint256(standardDebtSize * 2);
+        vm.prank(admin); vat.file(ilk, "line", debtCeiling * RAY);
+        hub.exec(ilk);
+        (ink, art) = vat.urns(ilk, address(pool));
+        assertEq(ink, debtCeiling);
+        assertEq(art, debtCeiling);
+        assertRoundingEq(getLPTokenBalanceInAssets(address(pool)), debtCeiling);
+    }
+
+    function test_collect_interest() public {
+        adjustDebt(standardDebtSize);
+
+        generateInterest();
+
+        uint256 vowDai = vat.dai(address(vow));
+        hub.exec(ilk);
+
+        assertGt(vat.dai(address(vow)), vowDai);
+    }
+
+    function test_insufficient_liquidity_for_unwind_fees() public {
+        uint256 currentLiquidity = getLiquidity();
+        uint256 vowDai = vat.dai(address(vow));
+
+        // Inrease debt
+        adjustDebt(standardDebtSize);
+
+        uint256 pAssets = getLPTokenBalanceInAssets(address(pool));
+        (uint256 pink, uint256 part) = vat.urns(ilk, address(pool));
+        assertEq(pink, part);
+        assertRoundingEq(pink, pAssets);
+
+        // Someone else borrows the exact amount previously available
+        (uint256 amountSupplied,) = vat.urns(ilk, address(pool));
+        uint256 amountToBorrow = currentLiquidity;
+        adjustLiquidity(-int256(amountToBorrow));
+
+        // Accumulate interest
+        generateInterest();
+
+        uint256 feesAccrued = getLPTokenBalanceInAssets(address(pool)) - pAssets;
+        currentLiquidity = getLiquidity();
+
+        assertGt(feesAccrued, 0);
+        assertEq(amountSupplied, currentLiquidity);
+        assertGt(amountSupplied + feesAccrued, currentLiquidity);
+
+        // Cage the system to trigger only unwinds
+        vm.prank(admin); hub.cage(ilk);
+        hub.exec(ilk);
+
+        uint256 assets = getLPTokenBalanceInAssets(address(pool));
+        // All the fees are accrued but what can't be withdrawn is added up to the original ink and art debt
+        (uint256 ink, uint256 art) = vat.urns(ilk, address(pool));
+        assertEq(ink, art);
+        assertRoundingEq(ink, assets);
+        assertGt(assets, 0);
+        assertRoundingEq(ink, pink + feesAccrued - currentLiquidity);
```

Comment:
```suggestion
        uint256 amountToBorrow = currentLiquidity;
        adjustLiquidity(-int256(amountToBorrow));

        // Accumulate interest
        generateInterest();

        uint256 feesAccrued = getLPTokenBalanceInAssets(address(pool)) - pAssets;
        currentLiquidity = getLiquidity();

        assertGt(feesAccrued, 0);
        assertEq(pink, currentLiquidity);
        assertGt(pink + feesAccrued, currentLiquidity);

        // Cage the system to trigger only unwinds
        vm.prank(admin); hub.cage(ilk);
        hub.exec(ilk);

        uint256 assets = getLPTokenBalanceInAssets(address(pool));
        // All the fees are accrued but what can't be withdrawn is added up to the original ink and art debt
        (uint256 ink, uint256 art) = vat.urns(ilk, address(pool));
        assertEq(ink, art);
        assertRoundingEq(ink, assets);
        assertGt(assets, 0);
        assertRoundingEq(ink, feesAccrued);
```

Reply:
Basically the change is to remove `amountSupplied` as it is the same than `pink` and also simplify the last assert which equals to `feesAccrued ` no need to sum and sub the same value (and it was actually pretty confusing).

Reply:
The plan is at some point we migrate the V2 tests to use this base, right?